pkg/apis/core/types_shoot.go

```diff
@@ -765,6 +765,10 @@ const (
 // KubeletConfig contains configuration settings for the kubelet.
 type KubeletConfig struct {
 	KubernetesConfig
+	// A quantity defines the maximum size of the container log file before it is rotated. For example: "5Mi" or "256Ki".
+	ContainerLogMaxSize *resource.Quantity
+	// Maximum number of container log files that can be present for a container.
+	ContainerLogMaxFiles *int32
```

Comment:
There is no validation for the newly introduced fields. For example I can set 
```yaml
          kubelet:
            containerLogMaxFiles: -1
```

which makes kubelet fail to start with 

```
E1014 13:17:19.087533 1254309 server.go:294] "Failed to run kubelet" err="failed to run Kubelet: failed to create kubelet: failed to initialize container log manager: invalid MaxFiles -1, must be > 1"
```

We usually add validation to prevent such cases to improve the end users and operators lives.

Reply:
yep let me add a check